### PR TITLE
[SWATANG-05] - add sign up feature

### LINKED ISSUE #5: Implement Signup feature

### PR DESCRIPTION
**Pull request by mwd-luongmai**
### Description
- Implement new feature to allow register new user
- Implement unit test for feature component
- Fixing using RegExp()

### ☑️ Definition of Done Checklist
- [x] Keep the application can be launched successfully
- [x] Keep the source code is built successfully
- [x] Keep the unit test can run successfully

### Check the box when resolved these items and provide more information

- [ ] Documentation
- [ ] Localization
- [x] Coding
- [x] Unit Tests
- [ ] Integration Tests
- [ ] System Tests

### 📷 Screenshot / GIF

Screen Shot 2021-06-23 at 11 57 32

Screen Shot 2021-06-23 at 11 59 16

Screen Shot 2021-06-23 at 11 59 28

### CODE REVIEW

File: src/app/core/services/authentication/authentication.service.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ import { ApiService } from '../api/api.service'
 const routes = {
   login: '/users/authenticate',
   logout: '/users/logout',
+  signup: '/users/register'
```
Comment (eisen-develops) [inline src/app/core/services/authentication/authentication.service.ts:12]: @mwd-luongmai are we using `/user` or `/users`? The previous version used users, but I know in the chat you said `user`

Comment (mwd-luongmai) [reply src/app/core/services/authentication/authentication.service.ts:12]: hi @1ceb3rg, this is the api route , not Angular route. Hope you don't confuse about that

**Comment by quangloc1993** [comment]
Looks good to merge.

**Review by quangloc1993 — APPROVED** [review]
Looks good to merge

**Review by eisen-develops — APPROVED** [review]

**Pull request merged**